### PR TITLE
Fixes #9138 - Intermittent HostTest/NIC integration test failures

### PR DESCRIPTION
**Pull request by tstrachota**
**Please don't merge yet.**
We need to run jenkins test several times to make sure this change helped. Unfortunately I'm very rarely able to reproduce the issue locally.

### CODE REVIEW

File: test/integration/host_test.rb [hunk 1]
```diff
@@ -117,6 +117,8 @@ def assert_interface_change(change, &block)
           assert modal.find('.interface_provision').checked?, "Provision checkbox is checked"
 
           modal.find(:button, "Cancel").click
+          # wait until modal is closed
+          page.has_css?('#interfaceModal', :visible => false)
```
Comment (domcleal) [inline test/integration/host_test.rb]: does this actually cause a wait/block somehow?  I'm not very familiar with capybara, but would assume it'd be similar to the table.find line below anyway.

Comment (tstrachota) [inline test/integration/host_test.rb]: AFAIK capybara should wait until such element exist or time out. The table already exist by the time the modal is being closed so it doesn't block and directly checks the content (my assumption).

**Comment by domcleal** [comment]
The update looks fine too, could you let me know how your test runs go please?

**Comment by tstrachota** [comment]
@domcleal we had one failure out of 10 test runs again. I've checked Capybara's codebase to make sure it waits as I expect it to wait. I've now updated the js to close the modal window as a very last step. It's the last in my mind that could be causing the problem. Running the tests again.

**Comment by domcleal** [comment]
If it's a 1/10 failure rate, you might want to run it 20 times, just in case :8ball:

**Comment by tstrachota** [comment]
@domcleal I've run out of ideas :-/. I've created https://github.com/theforeman/foreman/pull/2117 (going to test this one too) to unblock merges and get some time for fixing it.

**Comment by domcleal** [comment]
Incidentally, I tried increasing the Capybara wait time to 10 seconds on develop and re-ran tests, but they still failed reliably.  Maybe that wasn't enough, or there are sequences (fixed in this PR?) that aren't triggering the wait correctly.

I think this has gotten worse since Poltergeist was added.

**Comment by domcleal** [comment]
Did you want to re-open this PR @tstrachota?  It still needs fixing unfortunately, the tests are currently being skipped.

**Comment by tstrachota** [comment]
@domcleal I know there's still some work to be done. I deleted the branch by mistake and that closed the PR.

**Comment by domcleal** [comment]
Yep, will just re-open it then so we don't forget it's there.

**Comment by tbrisker** [comment]
@tstrachota,
Thank you very much for your efforts!
This PR has been open for a very long time with no activity.
Please indicate if you plan to continue working on it or not; if not, feel free to close it.
If you do not respond either way within a couple of weeks, I will assume this PR is abandoned and will go ahead and close it for you.

**Comment by domcleal** [comment]
These tests were re-enabled and fixed via 26dade0.

**Pull request closed without merging**